plugin-desktopswitch/desktopswitch.cpp

```diff
@@ -46,8 +46,9 @@ DesktopSwitch::DesktopSwitch(const ILxQtPanelPluginStartupInfo &startupInfo) :
     ILxQtPanelPlugin(startupInfo),
     m_pSignalMapper(new QSignalMapper(this)),
     m_desktopCount(KWindowSystem::numberOfDesktops()),
-    mRows(1),
-    mDesktops(new NETRootInfo(QX11Info::connection(), NET::NumberOfDesktops | NET::CurrentDesktop | NET::DesktopNames, NET::WM2DesktopLayout))
+    mRows(settings()->value("rows", 1).toInt()),
```

Comment:
Yes, I did that to be able to remove the `settingsChanged` call from `setup`, because now I had to call setup from `settingsChanged`. But I can probably do the same thing with all the configuration code in `settingsChanged`.